provider/lxd/environ.go

```diff
@@ -50,6 +53,19 @@ type Server interface {
 	CreateVolume(pool, name string, config map[string]string) error
 	UpdateStoragePoolVolume(pool string, volType string, name string, volume lxdapi.StorageVolumePut, ETag string) error
 	DeleteStoragePoolVolume(pool string, volType string, name string) (err error)
+	ServerCertificate() string
+}
+
+// ServerFactory creates a new factory for creating servers that are required
+// by the server.
+type ServerFactory interface {
```

Comment:
This should be co-located with the implementation. 

If you want to keep them together with the go:generate directive, I think it is worth creating a new file with Server, ServerFactory and the serverFactory implementation.

Reply:
agreed.